src/main/java/org/opensearch/knn/index/codec/params/KNNVectorsFormatParams.java

```diff
@@ -17,10 +18,16 @@
 public class KNNVectorsFormatParams {
     private int maxConnections;
     private int beamWidth;
+    private final SpaceType spaceType;
 
     public KNNVectorsFormatParams(final Map<String, Object> params, int defaultMaxConnections, int defaultBeamWidth) {
+        this(params, defaultMaxConnections, defaultBeamWidth, SpaceType.UNDEFINED);
+    }
+
+    public KNNVectorsFormatParams(final Map<String, Object> params, int defaultMaxConnections, int defaultBeamWidth, SpaceType spaceType) {
         initMaxConnections(params, defaultMaxConnections);
         initBeamWidth(params, defaultBeamWidth);
+        this.spaceType = spaceType;
     }
```

Comment:
do we need both of these constructors? can we not just keep 1.

Reply:
I didn't want to modify all the places the first constructor was being used so I'm basically overloading it like so.

Reply:
Yeah I understand. I will less code here by just having 1 constructor, and the old constructor I feel would be just getting used in tests so I feel it is safe to change the code there. 

But since this is not a make or break feature, I will leave this upto your best judgement.